hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -0,0 +1,308 @@
+			Serializable id,
+			SharedSessionContractImplementor session)
+			throws HibernateException {
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowInsertEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( LOG.isDebugEnabled() ) {
+			LOG.debugf(
+					"Inserting collection: %s",
+					MessageHelper.collectionInfoString( this, collection, id, session )
+			);
+		}
+
+		CompletionStage<Integer> loop = CompletionStages.zeroFuture();
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		collection.preInsert( this );
+
+		return CompletionStages.total(
+				entryArray(collection),
+				entry -> loop.thenCompose( total -> reactiveConnection
+								.update( getSQLInsertRowString(), paramValues( entry, collection, id,session ))
+								.thenApply( insertCount -> total + insertCount ) ) )
+				.thenAccept( total -> LOG.debugf( "Done inserting rows: %s inserted", total ) );
+	}
+
+	/**
+	 * @see org.hibernate.persister.collection.AbstractCollectionPersister#remove(Serializable, SharedSessionContractImplementor)
+	 */
+	public CompletionStage<Void> removeReactive(Serializable id, SharedSessionContractImplementor session)
+			throws HibernateException {
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+		if ( !isInverse && isRowDeleteEnabled() ) {
+			if ( LOG.isDebugEnabled() ) {
+				LOG.debugf(
+						"Deleting collection: %s",
+						MessageHelper.collectionInfoString( this, id, getFactory() )
+				);
+			}
+
+			List<Object> params = new ArrayList<>();
+			params.add( id );
+			String sql = getSQLDeleteString();
+			return CompletionStages.voidFuture()
+					.thenCompose( s -> reactiveConnection.update( sql, params.toArray( new Object[0] )))
+					.thenCompose(CompletionStages::voidFuture);
+		}
+		return CompletionStages.voidFuture();
+	}
+
+	/**
+	 * @see org.hibernate.persister.collection.AbstractCollectionPersister#deleteRows(PersistentCollection, Serializable, SharedSessionContractImplementor)
+	 */
+	@Override
+	public CompletionStage<Void> reactiveDeleteRows(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session) throws HibernateException {
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowDeleteEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		CompletionStage<Integer> loop = CompletionStages.zeroFuture();
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		boolean deleteByIndex = hasIndex && !indexContainsFormula;
+
+		return CompletionStages.total(
+				entryArray( collection, deleteByIndex ),
+				entry -> loop.thenCompose( total -> reactiveConnection
+						.update( getSQLDeleteRowString(), paramValues( entry, id ,session, deleteByIndex ) )
+						.thenApply( deleteCount -> total + deleteCount ) ) )
+				.thenAccept( total -> LOG.debugf( "Done removing rows: %s removed", total ) );
```

Comment:
return CompletionStages
				.total( entryArray( collection, deleteByIndex ),
						entry -> reactiveConnection.update( getSQLDeleteRowString(), paramValues( entry, id ,session, deleteByIndex ) ) )
				.thenAccept( total -> LOG.debugf( "Done removing rows: %s removed", total ) );

Reply:
You can remove entryArray if you add a new `total` method in `CompletionStages` accepting an iterable